Marlin/src/HAL/STM32/pinsDebug.h

```diff
@@ -280,9 +296,36 @@ void pwm_details(const pin_t Ard_num) {
         case 13 : SERIAL_ECHOPGM("DCMI"); break;
         case 14 : SERIAL_ECHOPGM("unused (shouldn't see this)"); break;
         case 15 : SERIAL_ECHOPGM("EVENTOUT"); break;
+#endif
       }
     }
   #else
     // TODO: F1 doesn't support changing pins function, so we need to check the function of the PIN and if it's enabled
   #endif
 } // pwm_details
+
+
+#ifdef STM32F746xx
+
+  int8_t digital_pin_to_analog_pin(const pin_t Ard_num) {
+    if (WITHIN(Ard_num, 0, 7))
+      return Ard_num - NUM_ANALOG_FIRST;
+    if (WITHIN(Ard_num, 16, 17))
+      return Ard_num - 8;
+    if (WITHIN(Ard_num, 32, 37))
+      return Ard_num - 22;
+    if (WITHIN(Ard_num, 83, 90))
+      return Ard_num - 67;
+    return -1;
+  }
```

Comment:
Double-checking the datasheet becuse this isn't exactly matching up…
`pinsDebug.h`:
```cpp
if (WITHIN(Ard_num, 0, 7))    return Ard_num;       // Pins  0 -  7  =>  A0  - A7
if (WITHIN(Ard_num, 16, 17))  return Ard_num - 8;   // Pins 16 - 17  =>  A8  - A9
if (WITHIN(Ard_num, 32, 37))  return Ard_num - 22;  // Pins 32 - 37  =>  A10 - A15
if (WITHIN(Ard_num, 83, 90))  return Ard_num - 67;  // Pins 80 - 90  =>  A16 - A23
```
`HAL.h`:
```cpp
if (WITHIN(Ard_num, 0, 7))    return Ard_num;        // A0  - A7   =>  Pins  0 -  7
if (WITHIN(Ard_num, 8, 9))    return Ard_num + 8;    // A8  - A9   =>  Pins 16 - 17
if (WITHIN(Ard_num, 11, 12))  return Ard_num + 22;   // A11 - A12  =>  Pins 33 - 34
if (WITHIN(Ard_num, 13, 23))  return Ard_num + 67;   // A13 - A23  =>  Pins 80 - 90
```
It looks like `pinsDebug.h` is probably the correct one since `HAL.h` is missing A10 altogether.

Reply:
Correction pushed.

Reply:
I'm thinking a different approach is needed.    Trying to produce a simple **digital_pin_to_analog_ch**  function for each variant will be like herding cats.  

I'm going to look at scanning the  **digitalPin** and **analogInputPin** arrays to see if the requested pin is available to be assigned as an AIN.

Worst comes to worst just looking at the MODER for the requested pin and returning true or false rather than a channel number may produce all that a M43 user is usually interested in.